omisego-sdk/src/main/java/co/omisego/omisego/websocket/channel/SocketChannel.kt

```diff
@@ -0,0 +1,186 @@
+package co.omisego.omisego.websocket.channel
+
+/*
+ * OmiseGO
+ *
+ * Created by Phuchit Sirimongkolsathien on 3/5/2018 AD.
+ * Copyright © 2017-2018 OmiseGO. All rights reserved.
+ */
+
+import co.omisego.omisego.model.socket.SocketSend
+import co.omisego.omisego.model.socket.SocketTopic
+import co.omisego.omisego.websocket.SocketChannelListener
+import co.omisego.omisego.websocket.SocketClientContract
+import co.omisego.omisego.websocket.SocketConnectionListener
+import co.omisego.omisego.websocket.SocketCustomEventListener
+import co.omisego.omisego.websocket.channel.SocketChannelContract.Dispatcher
+import co.omisego.omisego.websocket.channel.SocketChannelContract.SocketClient
+import co.omisego.omisego.websocket.channel.dispatcher.SocketDispatcherContract
+import co.omisego.omisego.websocket.enum.SocketEventSend
+import co.omisego.omisego.websocket.enum.SocketStatusCode
+import okhttp3.WebSocketListener
+
+/**
+ * A SocketChannel is responsible for handling join or leave from the socket channel.
+ *
+ * @param socketDispatcher A [Dispatcher] is responsible dispatch all events to the client.
+ * @param socketClient A [SocketClient] for sending a message to the eWallet web socket API and close the web socket connection.
+ */
+internal class SocketChannel(
+    override val socketDispatcher: SocketChannelContract.Dispatcher,
+    override val socketClient: SocketChannelContract.SocketClient
+) : SocketClientContract.Channel, SocketChannelContract.Channel, SocketDispatcherContract.SocketChannel {
+    private val channelSet: MutableSet<String> by lazy { mutableSetOf<String>() }
+
+    /**
+     * A [SocketMessageRef] is responsible for create unique ref value to be included in the [SocketSend] request.
+     */
+    override val socketMessageRef: SocketChannelContract.MessageRef = SocketMessageRef().apply { scheme = SocketMessageRef.SCHEME_JOIN }
+
+    /**
+     * An interval of milliseconds for scheduling the interval event such as the heartbeat event which used for keeping the connection alive.
```

Comment:
Is the heartbeat the only event concerned by this interval? If so why not name is `heartbeatInterval` or something like this?

(In any case, the doc sentence is a bit off, something like:
`An interval of milliseconds to schedule the interval event (heartbeat) used to keep the connection alive.`

Reply:
I was thinking of `reconnect` mechanism which is kind of `interval` thing, but it doesn't need for now because I haven't found any issue yet.

Reply:
Oh I see, then keeping it like this is good, just need to rephrase the doc :)